### PR TITLE
Schedule Editor

### LINKED ISSUE #8: Schedule pages
**Tasks involved**
- [x] Student side
- [ ] Class advisor side

**Figma design**
image

**Figma prototype**
[Link to Figma prototype](https://www.figma.com/proto/NF91V1N9blvG6slT68Z5bx/Our-SK?page-id=459%3A379&node-id=1065%3A2153&viewport=355%2C48%2C0.44&scaling=min-zoom&starting-point-node-id=989%3A302)

### PR DESCRIPTION
**Pull request by SiravitPhokeed**
<table>
  <tr>
    <td></td>
    <td>
      
      &nbsp;
      </td>
  </tr>
  <tr>
    <td></td>
    <td></td>
  </tr>
</table>

**Features**
- Schedule Editor
  Resolves #8
  - Schedule Period Hover Menu
    - Edit/delete button group
    - Drag handle (on pointer device only)
    - Extend handle
  - Drag and drop to move Period functionality
  - Drag edge to shorten/extend Period functionality
- Re-fetch button in Teacher Schedule
- Schedule now fetched from Supabase
  - Student Schedule
  - Teaching Schedule
  - Schedule Preview on Student Dashboard

**Fixes**
- Teachers link in Student Navigation is broken

### CODE REVIEW

File: public/static/locales/th/schedule.json [hunk 1]
```diff
@@ -1,12 +1,24 @@
...
+      "extend": "คลิกค้างแล้วลากเพื่อแก้ความยาว"
+    },
+    "extendGuide": {
+      "initial": "แตะ <1></1> เพื่อแก้ความยาวคาบนี้",
+      "tapToExtend": "แตะจุดที่จะให้ขอบขวาของคาบอยู่"
+    },
     "action": {
-      "add": "เพิ่มคาบสอน",
-      "edit": "แก้ไข"
+      "sync": "ซิงก์กับเซิร์ฟเวอร์",
```
Comment (ImSadudee) [inline public/static/locales/th/schedule.json:20]: This might be ซิงค์

**Review by ImSadudee — APPROVED** [review]
LGTM!

**Review by Jimmy-Tempest — APPROVED** [review]
LGTM!!!!

**Review by Temp9699 — APPROVED** [review]
LGTM

**Pull request merged**